### PR TITLE
add keyboard shortcuts

### PR DESCRIPTION
**Pull request by wangyira**
https://app.asana.com/0/1179618335628340/1196920778526702

- press enter to submit the form for adding, editing, deleting category/dish
- press esc to exit the form

### CODE REVIEW

File: src/components/dashboard/modal/category.js [hunk 1]
```diff
@@ -30,6 +35,17 @@ const NewCategoryModal = (props) => {
         }
     }
 
+    //press escape to exit the form, press enter to submit
+    function handler({ key }) {
+        if (ESCAPE_KEYS.includes(String(key))) {
+            props.toggleForm()
+        }
```
Comment (Tony1023) [inline src/components/dashboard/modal/category.js:42]: In what cases will `key` be a number and what cases will it be the string?

**Review by Tony1023 — APPROVED** [review]
Thanks @wangyira , the changes look great!

Just some small questions...
Are we plan to add these for the login page (and maybe the account) set up page?

File: src/components/dashboard/modal/category.js (hunk 1 above)
Comment (wangyira) [reply src/components/dashboard/modal/category.js]: I downloaded firefox and did more testing on chrome, safari, and firefox. Looks like we only need to use the key names such as Enter, and don't need both Enter and 13 (keyboardEventValue). Just updated the changes :)

**Comment by wangyira** [comment]
> Thanks @wangyira , the changes look great!
> 
> Just some small questions...
> Are we plan to add these for the login page (and maybe the account) set up page?

I was looking at the login page, I think it would make sense to login once the user press enter but they would probably click the sign up button to make a new account (no keyboard shortcut). I can add the code if you agree
Screen Shot 2020-11-01 at 11 58 05 PM

For the account set up, some pages are bidirectional, like this picture. Should they click **enter** to go to the next step and **left arrow** to go to the previous step? Or click **right arrow** to go to the next step and **left arrow** to go to the previous step?
Screen Shot 2020-11-01 at 11 53 45 PM

**Comment by Tony1023** [comment]
@wangyira 
LOL thanks for trying it on multiple browsers!

> I was looking at the login page, I think it would make sense to login once the user press enter but they would probably click the sign up button to make a new account (no keyboard shortcut). I can add the code if you agree

Yeah I think enter for login makes sense.

> For the account set up, some pages are bidirectional, like this picture. Should they click enter to go to the next step and left arrow to go to the previous step? Or click right arrow to go to the next step and left arrow to go to the previous step?

I thought about this again - I don't remember exactly what's on there but if there are any *multi-line input field*, then we probably don't want enter for next step. Otherwise I think enter for next step makes sense - no shortcut for previous step (arrow shortcuts would mess up typing :P)

**Comment by wangyira** [comment]
Thanks for the input! I checked the user set up flow - there's no multi-line input, so I added press enter to login and to navigate to the next page for setting up a new user

**Review by Tony1023 — APPROVED** [review]

**Pull request merged**